rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -488,6 +489,12 @@ bool PlayerImpl::play()
             RCLCPP_INFO_STREAM(owner_->get_logger(), "Sleep " << delay.nanoseconds() << " ns");
             std::chrono::nanoseconds delay_duration(delay.nanoseconds());
             std::this_thread::sleep_for(delay_duration);
+            if (!play_options_.start_paused) {
+              clock_->resume();
+            }
```

Comment:
I can predict that after some time someone will complain and raise a new issue that when using a player with `--loop` and `--start-paused` options and sending the pause service call request to the player somewhere at the end of the playback the request is ignored and player still will continue to playback from the beginning.
And we will have a hard time to debug this race condition, because of this `resume` call.
cc: @fujitatomoya @hayato-m126